lib/pure/strformat.nim

```diff
@@ -527,8 +527,13 @@ proc format*(value: SomeFloat; specifier: string; res: var string) =
   var sign = false
   if value >= 0.0:
     if spec.sign != '-':
-      f = spec.sign & f
       sign = true
+      if  value == 0.0:
+        if 1.0 / value == Inf:
+          # only add the sign if value != negZero
+          f = spec.sign & f
```

Comment:
Could you use `insert` instead, to modify 'f' in place? It would be nice to reduce the number of temporary string allocations made in this procedure.

Reply:
There are other places where we could use `insert` instead of string concatenation, but i limit the change strictly to the area i want to fix (no `+` in front of negative zero).

Personally i don't like inplace mutations, therefore I will not produce PR's to change other places in strformat to use inplace mutations.

Reply:
It doesn't matter if you like them or not, they are faster and this is a "standard library" where everybody benefits from faster code.